frontend/src/theme/utils.ts

```diff
@@ -201,26 +205,59 @@ export const createBaseUiTheme = (
     createThemeOverrides(theme)
   )
 
+export const createEmotionColors = (genericColors: {
+  [key: string]: string
+}): { [key: string]: string } => ({
+  ...genericColors,
+  // Alerts
+  alertErrorBorderColor: transparentize(genericColors.red, 0.8),
+  alertErrorBackgroundColor: transparentize(genericColors.red, 0.8),
+  alertErrorTextColor: genericColors.danger,
+  alertInfoBorderColor: transparentize(genericColors.blue, 0.9),
+  alertInfoBackgroundColor: transparentize(genericColors.blue, 0.9),
+  alertInfoTextColor: genericColors.info,
+  alertSuccessBorderColor: transparentize(genericColors.green, 0.8),
+  alertSuccessBackgroundColor: transparentize(genericColors.green, 0.8),
+  alertSuccessTextColor: genericColors.success,
+  alertWarningBorderColor: transparentize(genericColors.yellow, 0.2),
+  alertWarningBackgroundColor: transparentize(genericColors.yellow, 0.8),
+  alertWarningTextColor: genericColors.warning,
+
+  codeTextColor: genericColors.green80,
+  codeHighlightColor: genericColors.secondaryBg,
+
+  docStringHeaderBorder: "#e6e9ef",
+  docStringModuleText: "#444444",
+  docStringContainerBackground: "#f0f3f9",
+
+  headingColor: genericColors.bodyText,
+
+  tableGray: genericColors.gray40,
+})
+
 const createEmotionTheme = (
-  themeInput: Partial<CustomThemeConfig>,
+  themeInput: Partial<ICustomThemeConfig>,
   baseThemeConfig = baseTheme
 ): Theme => {
-  const { font, ...customColors } = themeInput
+  const { name, font, ...customColors } = themeInput
   // Mapping from CustomThemeConfig to color primitives
   const {
     secondaryBackground: secondaryBg,
     backgroundColor: bgColor,
     ...paletteColors
   } = customColors
-  const { colors, genericFonts } = baseThemeConfig.emotion
+  const { genericColors, genericFonts } = baseThemeConfig.emotion
+  const newGenericColors = {
+    ...genericColors,
+    ...(paletteColors as { [key: string]: string }),
```

Comment:
Huh I'm surprised this typecast ends up being necessary. Is it due to some weirdness with typescript not being able to infer types from the proto? I would have expected it to work without since I'm fairly sure it would if `customColors` were a plain js object

Reply:
Typescript thinks they can be `string | undefined` from the proto. I'm forcing them to be strings only since the optional undefined items should be stripped out from the destructuring